File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.c
#include <stdio.h>
#include <string.h>

int appendCharacters(char* s, char* t) {
    int x = 0, y = 0;
    while (x < strlen(s) && y < strlen(t)) {
        if (s[x] == t[y]) {
            y++;
        }
        x++;
    }
    
    if (y == strlen(t)) {
        return 0;
    }
    
    return strlen(t) - y;
}

File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.py
class Solution(object):
    def appendCharacters(self, s, t):
        x, y = 0, 0
        while x < len(s) and y < len(t):
            if s[x] == t[y]:
                y += 1
            x += 1
        
        if y == len(t):
            return 0
        
        return len(t) - y
